org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/proposals/AddImportCorrectionProposal.java

```diff
@@ -25,30 +23,20 @@
 import org.eclipse.ui.IEditorPart;
 
 import org.eclipse.jdt.core.ICompilationUnit;
-import org.eclipse.jdt.core.IJavaElement;
-import org.eclipse.jdt.core.IJavaProject;
-import org.eclipse.jdt.core.IModuleDescription;
-import org.eclipse.jdt.core.IPackageFragment;
-import org.eclipse.jdt.core.IPackageFragmentRoot;
-import org.eclipse.jdt.core.JavaModelException;
 import org.eclipse.jdt.core.dom.SimpleName;
 import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
-import org.eclipse.jdt.core.search.IJavaSearchConstants;
 
-import org.eclipse.jdt.internal.corext.util.JavaModelUtil;
-import org.eclipse.jdt.internal.corext.util.Messages;
 import org.eclipse.jdt.internal.corext.util.QualifiedTypeNameHistory;
 
 import org.eclipse.jdt.ui.text.java.correction.ASTRewriteCorrectionProposal;
-import org.eclipse.jdt.ui.text.java.correction.ASTRewriteCorrectionProposalCore;
-
-import org.eclipse.jdt.internal.ui.text.correction.CorrectionMessages;
-import org.eclipse.jdt.internal.ui.text.correction.ModuleCorrectionsSubProcessor;
 
 public class AddImportCorrectionProposal extends ASTRewriteCorrectionProposal {
 
-	static String JAVA_BASE= "java.base"; //$NON-NLS-1$
-
+	static String JAVA_BASE= AddImportCorrectionProposalCore.JAVA_BASE;
```

Comment:
I'm seeing IllegalAccessError thrown from this line, because it tries to read a non-public field from a different classloader.

I no longer see any references to this class in JDT, so it must be my plugin that now tries to instantiate a dead class. IS IT DEAD?

Reply:
Yes.  The one original user of AddImportCorrectionProposal has replaced it with AddImportCorrectionProposalCore.

Reply:
Thanks, so should this class be removed now (and be it just to prevent others from running into that IllegalAccessError)?

Reply:
Yes, that's a good idea.  I'll open an issue for it.

Reply:
#1223 opened

Reply:
Didn't see Rob's PR so closed #1223